Assets/__Scripts/MapEditor/Input/BeatmapInputController.cs

```diff
@@ -74,11 +75,17 @@ public void OnQuickDelete(InputAction.CallbackContext context)
         RaycastFirstObject(out T obj);
         if (obj != null && context.performed)
         {
-            BeatmapObjectContainerCollection.GetCollectionForType(obj.objectData.beatmapType)
-                .DeleteObject(obj.objectData, true, true, "Deleted by the user.");
+            StartCoroutine(CompleteDelete(obj));
         }
     }
 
+    public IEnumerator CompleteDelete(T obj)
+    {
+        yield return null;
```

Comment:
Because the delete event gets fired on every object type and if we delete it immediately the raycast for the next type goes through the object you just deleted and deletes something else. So this means that we do all the decision work and then the coroutine deletes the thing that you actually wanted.